### DISCUSSION TITLE
Add something like HSTS for provenance

### DISCUSSION THREAD

**Discussion by Jack-Works** [npm/feedback#913] [Registry]
It should be a one-way opt-in setting for a package. Once it is opened, it cannot be closed, and all future packages must be published with provenance.

**Comment by feelepxyz** [comment]
💯  We're thinking of making provenance details "sticky" for a given package, so as soon as you publish with provenance, we bind the source repo URL and IDs to the package and prevent these details from changing or being removed in subsequent publishes unless you log in with 2FA and remove the provenance connection, allowing you to remove or change it.

I also like this idea of making provenance one-way opt-in, so you could never remove provenance for a given package unless you maybe get in touch with support and do some account verification.

**Reply by ljharb** [reply]
fwiw any sort of "one-way"-ness would make adopting provenance in my publishes an immediate nonstarter - it might be worth weighing the impact of that kind of restriction.

**Reply by feelepxyz** [reply]
> fwiw any sort of "one-way"-ness would make adopting provenance in my publishes an immediate nonstarter - it might be worth weighing the impact of that kind of restriction.

💯  yes would definitely think this through very carefully. If we did anything like this it would definitely be opt-in from maintainer and not make any difference to the consumer. I think it should be up to the maintainer to choose if they want stricter protections around publishing and are willing to accept the downsides. I think we should also get to the point were there are a lot of different options available for publishing with provenance.
